src/main/java/htsjdk/samtools/util/StringUtil.java

```diff
@@ -545,4 +545,56 @@ public static int levenshteinDistance(final String string1, final String string2
 
         return i;
     }
+
+    /**
+     * Calculates the hamming distance between two strings s1 and s2.
+     * Since Hamming distance is not defined for strings of differing lengths, we throw an exception if
+     * the two strings are of different lengths.
+     *
+     * @param s1 The first string to compare
+     * @param s2 The second string to compare, note that if s1 and s2 are swapped the value returned will be identical.
+     * @return Hamming distance between s1 and s2.
+     * @throws IllegalArgumentException If the two strings have differing lengths.
+     */
+    public static int hammingDistance(final String s1, final String s2) {
```

Comment:
@fleharty Sorry for the very slow response.  I think this note about N's is more confusing that it is clarifying.  I would say something more like "This implementation is case sensitive and does not have any special treatment for DNA."